datafusion/sql/tests/cases/plan_to_sql.rs

```diff
@@ -525,6 +525,60 @@ fn roundtrip_statement_with_dialect() -> Result<()> {
             parser_dialect: Box::new(GenericDialect {}),
             unparser_dialect: Box::new(SqliteDialect {}),
         },
+        TestStatementWithDialect {
+            sql: "SELECT * FROM UNNEST([1,2,3])",
+            expected: r#"SELECT * FROM UNNEST([1, 2, 3])"#,
+            parser_dialect: Box::new(GenericDialect {}),
+            unparser_dialect: Box::new(CustomDialectBuilder::default().with_unnest_as_table_factor(true).build()),
```

Comment:
the added tests all seem to use `with_unnest_as_table_factor(true)`
if the case of `with_unnest_as_table_factor(false)` already covered elsewhere?

Reply:
Nice catch. I added some cases for it in [954d76b](https://github.com/apache/datafusion/pull/13660/commits/954d76b4b4726792a62a29af39ff923615af2fe6)